otherlibs/systhreads/st_posix.h

```diff
@@ -166,128 +206,46 @@ Caml_inline int st_masterlock_waiters(st_masterlock * m)
 */
 Caml_inline void st_thread_yield(st_masterlock * m)
 {
+  uintnat waiters;
+
   pthread_mutex_lock(&m->lock);
   /* We must hold the lock to call this. */
-  assert(m->busy);
 
   /* We already checked this without the lock, but we might have raced--if
      there's no waiter, there's nothing to do and no one to wake us if we did
      wait, so just keep going. */
-  if (m->waiters == 0) {
+  waiters = atomic_load_acq(&m->waiters);
+
+  if (waiters == 0) {
     pthread_mutex_unlock(&m->lock);
     return;
   }
 
   m->busy = 0;
+  atomic_fetch_add(&m->waiters, +1);
   pthread_cond_signal(&m->is_free);
-  m->waiters++;
+  // releasing the domain lock but not triggering bt messaging
```

Comment:
Use `/* */` for comments.

Reply:
Fixed in https://github.com/ocaml-multicore/ocaml/commit/a8d0e5f2dd50cee50c63a80ef7aeab7df20eae32